jsx/brisk_ppx.ml

```diff
@@ -104,7 +99,14 @@ module Declaration_ppx = struct
                   [%expr [%e component_ident ~loc] ~key [%e child_expression]]
           | _ ->
               Location.raise_errorf ~loc
-                "A labelled argument or () was expected")
+                "A labelled argument or () was expected" )
+      | { pexp_desc = Pexp_newtype (ident, child_expression) } ->
+          {
+            expr with
+            pexp_desc =
+              Pexp_newtype (ident, map_component_expression child_expression);
```

Comment:
What is the error message (if any) if a user does:

```
let%component x = (type a, ~foo, type b) => ...
```

Should it be allowed? Does OCaml compiler throw an error? Are the locations correct?

Reply:
Oh, good question. That definitely seems like it should be both valid and allowed. It seems unlikely to occur in practice though, but I'll look more into it.

Reply:
Yup, ideally the current implementation is correct and OCaml compiler returns an error (if needed).

Reply:
It was accepted by both the PPX and compiler, so all good! I added this case to the test as well though.